src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -139,10 +140,30 @@ public void createAddGameFromRoom(Room room) {
         Game game = GameMapper.fromRoom(room);
         Long time = room.getDuration();
 
-        List<Problem> problems = problemService.getProblemsFromDifficulty(room.getDifficulty(), room.getNumProblems());
-        game.setProblems(problems);
-        setStartGameTimer(game, time);
+        // If specific problems specified, use those instead of difficulty setting
+        List<Problem> problems = game.getProblems();
+        problems.addAll(room.getProblems());
+
+        // Fill remaining problems with random ones by difficulty
+        int remaining = room.getNumProblems() - problems.size();
 
+        if (remaining < 0) {
+            throw new ApiException(RoomError.TOO_MANY_PROBLEMS);
+        }
+        if (remaining > 0) {
+            List<Problem> otherProblems = problemService.getProblemsFromDifficulty(room.getDifficulty(), room.getNumProblems());
+            for (Problem problem : otherProblems) {
+                if (!problems.contains(problem) && problems.size() < room.getNumProblems()) {
+                    problems.add(problem);
+                }
```

Comment:
You could add a `break;` statement  if `problems.size() >= room.getNumProblems()` so it doesn't iterate through all of the `otherProblems`; not crucial though, so I'll leave this up to you.

Reply:
whoops I accidentally added it within this if statement which obviously caused the tests to fail - since the break requires another if statement I think I might just leave it as is then